src/app/components/popup-host/popup-host.component.ts

```diff
@@ -0,0 +1,36 @@
+import {
+    AfterViewInit,
+    Component,
+    Input,
+    TemplateRef,
+    ViewChild,
+    ViewContainerRef,
+} from '@angular/core';
+
+@Component({
+    selector: 'app-popup-host',
+    templateUrl: './popup-host.component.html',
+    styleUrls: ['./popup-host.component.scss'],
+})
+export class PopupHostComponent implements AfterViewInit {
+    isVisible = false;
+
+    @Input() set template(template: TemplateRef<unknown> | null) {
+        this.viewContainer?.clear();
+        this.isVisible = false;
```

Comment:
2 раза в одном запуске сетера может быть присвоено значение `this.isVisible`: тут и ниже.
Кажется, в этом нет явной необходимости, можно в `isVisible` значнеие записать на основе `template`:
```suggestion
        this.isVisible = Boolean(template);
```

Reply:
поправила